buildSrc/src/main/groovy/io/vantiq/extsrc/assy/tasks/AssemblyGen.groovy

```diff
@@ -0,0 +1,422 @@
+import org.snakeyaml.engine.v2.api.LoadSettings
+import org.snakeyaml.engine.v2.api.Dump
+import org.snakeyaml.engine.v2.common.FlowStyle
+import org.gradle.api.Project
+
+import javax.inject.Inject
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.Paths
+import java.util.jar.JarFile
+import java.util.regex.Pattern
+
+/**
+ * Gradle task to build assembly specifications from Camel Kamelet definitions.
+ */
+
+@Slf4j
+class AssemblyGen extends DefaultTask {
+    public static final String KAMELET_GEN_BASE = 'vantiqGeneratedResources'
+    public static final String VANTIQ_SERVER_CONFIG = 'vantiq://server.config'
+    public static final String VANTIQ_SERVER_CONFIG_JSON = VANTIQ_SERVER_CONFIG + '?consumerOutputJson=true'
+    
+    public static final String KAMELETS_RESOURCE_FOLDER = 'kamelets'
+    public static final String KAMELETS_RESOURCE_PATH = KAMELETS_RESOURCE_FOLDER + '/'
+    public static final String YAML_FILE_TYPE = '.yaml'
+    public static final String YML_FILE_TYPE = '.yml'
+    public static final String SOURCE_KAMELET_PREFIX = '-source.kamelet'
+    public static final String SINK_KAMELET_PREFIX = '-sink.kamelet'
+    public static final String SOURCE_KAMELET_YAML_SUFFIX = SOURCE_KAMELET_PREFIX + YAML_FILE_TYPE
+    public static final String SOURCE_KAMELET_YML_SUFFIX = SOURCE_KAMELET_PREFIX + YML_FILE_TYPE
+    public static final String SINK_KAMELET_YAML_SUFFIX = SINK_KAMELET_PREFIX + YAML_FILE_TYPE
+    public static final String SINK_KAMELET_YML_SUFFIX = SINK_KAMELET_PREFIX + YML_FILE_TYPE
+
+    public static final String KAMELET_SOURCE = 'kamelet:source'
+    public static final String KAMELET_SINK = 'kamelet:sink'
+    
+    public static final String ASSEMBLY_PACKAGE_BASE = 'io.vantiq.extsrc.camel.kamelets'
+    public static final String PACKAGE_SEPARATOR = '.'
+    
+    // Vantiq directory names for project contents
+    public static final String VANTIQ_DOCUMENTS_DIR = 'documents'
+    public static final String VANTIQ_PROJECTS_DIR = 'projects'
+    public static final String VANTIQ_SOURCES_DIR = 'sources'
+    public static final String VANTIQ_TYPES_DIR = 'types'
+
+    public static final String CAMEL_CONNECTOR_SOURCE_TYPE = 'CAMEL_SOURCE'
+    public static final String PROPERTY_PLACEHOLDER_SUFFIX = '_placeholder'
+    public static final String CONFIG_CAMEL_RUNTIME = 'camelRuntime'
+    public static final String CONFIG_CAMEL_GENERAL = 'general'
+    public static final String CAMEL_RUNTIME_APPNAME = 'appName'
+    public static final String CAMEL_RUNTIME_ROUTE_DOCUMENT = 'routeDocument'
+    public static final String YAML_ROUTE_SUFFIX = '.routes.yaml'
+
+    public final Project project
+    
+    @Inject
+    AssemblyGen(Project project) {
+        this.project = project
+        if (project == null) {
+            throw new GradleException('AssemblyGen requires a non-null project.')
+        }
+    }
+    
+    @OutputDirectory
+    File getOutputDirectory() {
```

Comment:
Convert to more native use of gradle API (`getBuildDir()` instead of `.property('buildDir')`